### DISCUSSION TITLE
Jak 1 Camera Features and Improvements

### DISCUSSION THREAD

**Discussion by PussInScreen** [open-goal/jak-project#2524] [Ideas]
### What feature is your idea related to?

Based on the following camera related improvement ideas listed here;

- [FoV Sliders](https://github.com/open-goal/jak-project/issues/386#issuecomment-1161048602)
- [Centered Camera Toggle (Old-School Camera Zoom Vs. Tilt for right stick up/down movement)](https://github.com/open-goal/jak-project/issues/386#issuecomment-1161048602)
  - [Also mentioned here...](https://github.com/open-goal/jak-project/issues/386#issuecomment-1220129022)
- [Centre Camera Button / Input](https://github.com/open-goal/jak-project/issues/386#issuecomment-1215378038)
- [Character Based First Person Camera Toggle](https://github.com/open-goal/jak-project/issues/386#issuecomment-1215378038)
- [Move Camera Slightly In Locked Positions](https://github.com/open-goal/jak-project/issues/386#issuecomment-1220129022)
  - [Implementation suggestion made here that I agree with regarding tilting...](https://github.com/open-goal/jak-project/issues/386#issuecomment-1227690015)

Some Quick notes of my own though;

- This one is a matter of debate (as the games worked like this originally) but the normal vertical camera setting works such that pushing the right stick up pulls the camera out, by default it's set to inverted so that pushing the stick up pushes the camera towards Jak. I think the default behaviour is fine but calling this "Inverted" seems inaccurate. So maybe flip the definitions here.
- I find the camera controls via controller are kinda... I don't know... clunky and dated? It would be nice to have more linear scaling on the zoom / rotation and less/no acceleration maybe?
- In a similar vein to the above, adjustable sensitivity sliders for camera controls.
- Move camera options out of the currently nested state in game options and instead display them under graphic options, again, matter of debate, but I find camera settings are the most finicky things to get right on first adjustment and the less button presses to accessing them in the settings the better.
- Ability to turn off forced camera perspectives altogether. One place where this is a benefit is when you swim from Sandover Village to Sentinel Beach, around the rock arch, the camera will force itself to look at said arch to hide any pop-in / loading of the Sentinel Beach zone. I've found this zone loads so fast on NVME drives or other equivalent hardware that this forced perspective is no longer needed.

### Describe the solution you'd like.

Improve the camera as described, include the necessary toggles in the camera menu, I think some of these improvements should be set to default "on", such as the camera tilt for the locked camera, centre camera toggle (R3 was suggested) and the Zoom/Rotate at same time fix. I also think (if there's no objections), the zoom version of the third person camera should have the behaviours swapped between normal and inverted as well as just moving the camera settings up a level in the settings menu.

Things that should be off by default in my opinion are the disabling of the forced camera perspectives, the "new" centred camera and the first person camera starting point based on where Jak is facing as I can see these changes as being game breaking or going against the grain of the original vision. All sensitivity and FOV sliders should be set to values that align with the original game's camera by default.

[I'm going to start working on this myself here.](https://github.com/andrewk10/jak-project/tree/Camera_Features_and_Improvements)

In the meantime I'm happy to hear feedback and suggestions on how to do all this. Right now I'll just work on Jak 1 and once Jak 2 is in a more stable state I'll port as many of these improvements over as I can. I really wanted to contribute to such an amazing project and I see this as a good way of doing so. :)

Edit 1; I added the link to the branch on my fork for working on this, hadn't tried making any changes as of writing yet so no pull request yet.

Edit 2; I added the suggestion for disabling the forced camera perspective, elaborated on the changes which are off by default and on my own motivations.

Edit 3; Added checklist for progress tracking and adding screenshots for additional context, I won't be making notes of those edits here I'll just add a comment when a bit is finished, this section is long enough lol

#### Checklist of Stuff to Do
- [x] Move camera options into the main options and out of game options
- [ ] Implement FOV Sliders for third person and first person cameras
- [ ] Centred camera toggle to change the camera system from the legacy zoom in and out to a modern up and down movement
- [ ] Centre camera button to set camera to the original / initial co-ordinates at any given time
- [ ] Character based first person camera toggle
- [ ] Ability to move camera slightly in locked positions
- [x] Swap the definitions of normal and inverted for vertical third person camera movements
- [ ] Examine \(maybe rework) camera acceleration / smoothness, make less clunky maybe?
- [ ] Adjustable sensitivity for camera movements
- [ ] Ability to toggle off forced camera perspectives

#### Screenshots of Finished Items
![image](https://user-images.githubusercontent.com/19553807/220751728-cf048d9d-608a-4b7a-852d-145f262311cb.png)
_Fig 1: Move camera options into the main options and out of game options_

[Youtube Video Here](https://youtu.be/LLBtt6bx6ug)
_Mov 1: Swap the definitions of normal and inverted for vertical third person camera movements_

**Comment by PussInScreen** [comment]
Moving camera out of game options and having it in the main options complete, appears to be working :)

**Comment by PussInScreen** [comment]
Forgot to mention, I changed the definition of normal and inverted for the vertical third person camera the night before last

**Comment by Keith94** [comment]
FOV sliders would make the game much, much more playable; hope to see it.

**Comment by ALVEZ112** [comment]
Sad to see that this went nowhere :(
